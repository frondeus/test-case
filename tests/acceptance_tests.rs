#![cfg(test)]

mod acceptance {
    use itertools::Itertools;
    use std::env;
    use std::path::PathBuf;
    use std::process::Command;

    #[test]
    fn basic() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("basic"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }

    #[test]
    fn hamcrest_assertions() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("hamcrest_assertions"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }

    #[test]
    fn r#async() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("async"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }
}
