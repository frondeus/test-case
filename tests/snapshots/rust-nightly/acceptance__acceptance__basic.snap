---
source: tests/acceptance_tests.rs
expression: lines

---
running 0 tests
running 78 tests
test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
test result: ok. 75 passed; 0 failed; 3 ignored; 0 measured; 0 filtered out
test test_cases::abs_tests::returns_0_for_0 ... ok
test test_cases::abs_tests::returns_given_number_for_positive_input ... ok
test test_cases::abs_tests::returns_opposite_number_for_non_positive_input ... ok
test test_cases::arg_expressions::_2_4_6_to_string_expects ... ok
test test_cases::bar::_expects_string_default_ ... ok
test test_cases::basic_test::_1_expects ... ok
test test_cases::complex_tests::almost_eq1 ... ok
test test_cases::complex_tests::almost_eq2 ... ok
test test_cases::complex_tests::eq1 ... ok
test test_cases::complex_tests::eq2 ... ok
test test_cases::complex_tests::geq1 ... ok
test test_cases::complex_tests::geq2 ... ok
test test_cases::complex_tests::gt1 ... ok
test test_cases::complex_tests::gt2 ... ok
test test_cases::complex_tests::leq1 ... ok
test test_cases::complex_tests::leq2 ... ok
test test_cases::complex_tests::lt1 ... ok
test test_cases::complex_tests::lt2 ... ok
test test_cases::const_in_arg::this_is_desc_not_an_argument ... ok
test test_cases::contains_tests::vec_1_2_3_4_expects_complex_1 ... ok
test test_cases::contains_tests::vec_1_2_3_4_expects_complex_3_4_ ... ok
test test_cases::create_path::_cargo_toml_expects_complex_path ... ok
test test_cases::create_path::_src_lib_rs_expects_complex_file ... ok
test test_cases::create_path::long_dir ... ok
test test_cases::create_path::short_dir ... ok
test test_cases::divide_by_zero_f64_with_lambda::_0_0_expects_with_v_f64_assert_v_is_nan_ ... ok
test test_cases::divide_by_zero_f64_with_lambda::_1_0_expects_with_v_f64_assert_v_is_infinite_ ... ok
test test_cases::fancy_addition::some_2_3_some_4_expects_2_3_4 ... ok
test test_cases::fancy_addition::some_2_some_3_expects_5 ... ok
test test_cases::fancy_addition::treats_none_as_0 ... ok
test test_cases::impl_trait::_foo_expects ... ok
test test_cases::inconclusives::_expects_inconclusive_ ... ignored
test test_cases::inconclusives::inconclusive_test ... ignored
test test_cases::inconclusives::test_is_not_ran ... ignored
test test_cases::keyword_test::true_expects ... ok
test test_cases::leading_underscore_in_test_name::_dummy_expects ... ok
test test_cases::lowercase_test_name::_dummy_code_expects ... ok
test test_cases::multiple_test_cases::_2_expects ... ok
test test_cases::multiple_test_cases::_4_expects ... ok
test test_cases::multiplication_tests::when_both_operands_are_negative ... ok
test test_cases::multiplication_tests::when_both_operands_are_possitive ... ok
test test_cases::multiplication_tests::when_operands_are_swapped ... ok
test test_cases::name::_2nd_test ... ok
test test_cases::name::_4_5_expects ... ok
test test_cases::name::test_1_8_9 ... ok
test test_cases::name::test_3_6_9 ... ok
test test_cases::nested::nested_test_case::_1_1_expects ... ok
test test_cases::nested::using_fn_from_super::_20_22_expects ... ok
test test_cases::nested::using_fn_from_super::_42_expects ... ok
test test_cases::not_complex::_1_0_expects_complex_not_almost_2_1_p_0_01 ... ok
test test_cases::not_complex::_1_0_expects_complex_not_eq_2_5 ... ok
test test_cases::not_complex::_1_0_expects_complex_not_not_not_not_lt_3_0 ... ok
test test_cases::not_contains_tests::vec_1_2_3_4_expects_complex_not_3_2_ ... ok
test test_cases::not_contains_tests::vec_1_2_3_4_expects_complex_not_5 ... ok
test test_cases::not_path::_cargo_toml_parse_unwrap_expects_complex_not_dir ... ok
test test_cases::not_path::_cargo_yaml_parse_unwrap_expects_complex_not_path ... ok
test test_cases::not_path::_src_parse_unwrap_expects_complex_not_file ... ok
test test_cases::panicing::_expects_panicking_some_it_has_to_panic_ - should panic ... ok
test test_cases::panics_without_value::_expects_panicking_none - should panic ... ok
test test_cases::pattern_matching_result::simpleenum_var2_expects_matching_simpleenum_var2 ... ok
test test_cases::pattern_matching_result_fails::simpleenum_var1_expects_matching_simpleenum_var2 - should panic ... ok
test test_cases::power_of_two_with_using::_1_expects_use_assert_is_power_of_two ... ok
test test_cases::power_of_two_with_using::_2_expects_use_crate_test_cases_assert_is_power_of_two ... ok
test test_cases::power_of_two_with_using::_4_expects_use_some_mod_assert_is_power_of_two ... ok
test test_cases::qualified_attribute::first_test ... ok
test test_cases::qualified_attribute::second_test ... ok
test test_cases::result::_2_expects_4 ... ok
test test_cases::result::_4_expects_8 ... ok
test test_cases::result_and_name::_4_5_expects_9 ... ok
test test_cases::result_and_name::test_no_1 ... ok
test test_cases::result_expresion_with_name::test_result_expression ... ok
test test_cases::result_expression::_2_2_expects_2_2 ... ok
test test_cases::result_which_panics::_2_2_expects_2_3 - should panic ... ok
test test_cases::result_with_mod_sep::_42_expects_std_string_string_new_ ... ok
test test_cases::test_generics::source1_expects_1 ... ok
test test_cases::test_generics::source2_expects_2 ... ok
test test_cases::test_impl::source1_expects_1 ... ok
test test_cases::test_impl::source2_expects_2 ... ok
